File: app/src/main/java/de/michael/filebinmobile/fragments/NavigationFragment.java
package de.michael.filebinmobile.fragments;

import android.support.v4.app.Fragment;

import de.michael.filebinmobile.OnTabNavigationRequestedListener;

public class NavigationFragment extends Fragment {

    private OnTabNavigationRequestedListener onTabNavigationRequestedListener;

    public void setOnTabNavigationRequestedListener(OnTabNavigationRequestedListener onTabNavigationRequestedListener) {
        this.onTabNavigationRequestedListener = onTabNavigationRequestedListener;
    }

    OnTabNavigationRequestedListener getOnTabNavigationRequestedListener() {
        return onTabNavigationRequestedListener;
    }
}

File: app/src/main/java/de/michael/filebinmobile/fragments/refactor/NavigationFragment.kt
package de.michael.filebinmobile.fragments.refactor

import android.support.v4.app.Fragment


abstract class NavigationFragment : Fragment() {
    var onTabNavigationRequested: (menuItemId: Int) -> Unit = {}

    abstract fun cancelAllPossiblyRunningTasks()

    // first called in lifecycle if fragment is replaced or removed
    override fun onPause() {
        cancelAllPossiblyRunningTasks()
        super.onPause()
    }
}